android/jni/com/mapswithme/maps/Sponsored.cpp

```diff
@@ -119,43 +117,37 @@ JNIEXPORT void JNICALL Java_com_mapswithme_maps_widget_placepage_Sponsored_nativ
   string const hotelId = jni::ToNativeString(env, id);
   string const code = jni::ToNativeString(env, locale);
 
-  g_framework->RequestBookingInfo(hotelId, code, [hotelId](
-                                                     BookingApi::HotelInfo const & hotelInfo) {
-    GetPlatform().RunOnGuiThread([=]() {
-      JNIEnv * env = jni::GetEnv();
-
-      auto description = jni::ToJavaString(env, hotelInfo.m_description);
-      auto photos =
-          jni::ToJavaArray(env, g_imageClass, hotelInfo.m_photos,
-                           [](JNIEnv * env, BookingApi::HotelPhotoUrls const & item) {
-                             return env->NewObject(g_imageClass, g_imageConstructor,
-                                                   jni::ToJavaString(env, item.m_original),
-                                                   jni::ToJavaString(env, item.m_small));
-                           });
-      auto facilities =
-          jni::ToJavaArray(env, g_facilityTypeClass, hotelInfo.m_facilities,
-                           [](JNIEnv * env, BookingApi::Facility const & item) {
-                             return env->NewObject(g_facilityTypeClass, g_facilityConstructor,
-                                                   jni::ToJavaString(env, item.m_id),
-                                                   jni::ToJavaString(env, item.m_localizedName));
-                           });
-      auto reviews = jni::ToJavaArray(
-          env, g_reviewClass, hotelInfo.m_reviews,
-          [](JNIEnv * env, BookingApi::HotelReview const & item) {
-            return env->NewObject(
-                g_reviewClass, g_reviewConstructor, jni::ToJavaString(env, item.m_reviewNeutral),
-                jni::ToJavaString(env, item.m_reviewPositive),
-                jni::ToJavaString(env, item.m_reviewNegative),
-                jni::ToJavaString(env, item.m_author), jni::ToJavaString(env, item.m_authorPictUrl),
-                item.m_rating,
-                time_point_cast<milliseconds>(item.m_date).time_since_epoch().count());
-          });
-      auto nearby = env->NewObjectArray(0, g_nearbyObjectClass, 0);
-
-      env->CallStaticVoidMethod(g_sponsoredClass, g_infoCallback, jni::ToJavaString(env, hotelId),
-                                env->NewObject(g_hotelInfoClass, g_hotelInfoConstructor,
-                                               description, photos, facilities, reviews, nearby));
-    });
+  g_framework->RequestBookingInfo(hotelId, code, [=](booking::HotelInfo const & hotelInfo) {
+    JNIEnv * env = jni::GetEnv();
+
+    auto description = jni::ToJavaString(env, hotelInfo.m_description);
+    auto photos = jni::ToJavaArray(env, g_imageClass, hotelInfo.m_photos,
+                                   [](JNIEnv * env, booking::HotelPhotoUrls const & item) {
```

Comment:
Could you please create a named functions, say,
```c++
jobject ToPhotoObject(JNIEnv * env, booking::HotelPhotoUrls const & item);
...
```
and use them here, instead of these lambdas?